### DISCUSSION TITLE
希望文件管理的打开文件对话框支持输入文件全路径

### DISCUSSION THREAD

**Discussion by Ernest-su** [linuxdeepin/developer-center#3400] [Features Request & Ideas | 特性请求 & 头脑风暴]
![录屏_google-chrome_20220830154423](https://user-images.githubusercontent.com/5917446/187395821-b46059b9-1fa5-494e-afb6-44e180532646.gif)

**Comment by Ernest-su** [comment]
复现步骤：
1. 打开百度，点搜索框右边的图片图标
2. 打开文件
3. 在打开的【打开文件页面 按ctrl+L，在地址栏输入文件地址
4. 按回车，无法定位到文件
5. 去掉文件名，按回车，能定位到文件所在文件夹。

**Comment by BLumia** [comment]
cc @meiyixiang

**Comment by Ernest-su** [comment]
https://github.com/linuxdeepin/dde-file-manager/blob/88fb76c3cf9bf48009c4b04ebccd494d38defded/src/dde-file-manager-lib/views/dfiledialog.cpp#L234    这里cd前判断url是文件的全路径，截取父目录的路径，应该就能够实现定位到文件所在文件夹的功能了。
https://github.com/linuxdeepin/dde-file-manager/blob/88fb76c3cf9bf48009c4b04ebccd494d38defded/src/dde-file-manager-lib/views/dfileview.cpp#L642
https://github.com/linuxdeepin/dde-file-manager/blob/88fb76c3cf9bf48009c4b04ebccd494d38defded/src/dde-file-manager-lib/views/dfilemanagerwindow.cpp#L950
这个几个 都是用的cd。缺少对 指向文件的路径的判断

**Reply by BLumia** [reply]
实现肯定能实现，但需求还是需要先确认应不应该去实现，然后再考虑做的 😂

**Comment by Ernest-su** [comment]
![录屏_VirtualBox Machine_20220831104453](https://user-images.githubusercontent.com/5917446/187581849-1a4b22f7-cc3b-460d-8b25-e773b7f692e9.gif)
deepin的文件管理器原来已经支持这个功能了。 只是UOS的还没跟上，或者裁剪了

**Reply by BLumia** [reply]
你录屏的这个调起的文件打开对话框并非深度文件管理器提供的= =

**Reply by Ernest-su** [reply]
你是指第一个录屏的UOS版本，不是深度文件管理器提供的，还是指第二个录屏的deepin版本，不是深度文件管理器提供的？

**Reply by BLumia** [reply]
> 你是指第一个录屏的UOS版本，不是深度文件管理器提供的，还是指第二个录屏的deepin版本，不是深度文件管理器提供的？

https://github.com/orgs/linuxdeepin/discussions/3400#discussioncomment-3514043 （就是我直接回复的这层）

**Reply by Ernest-su** [reply]
![截图_选择区域_20220831114213](https://user-images.githubusercontent.com/5917446/187588125-4d793e8a-669e-47f0-8e6a-34c9aaf7b005.png)
界面跟系统内置的文件管理器长得一模一样，而且所有用到打开文件的地方，都是这个界面。

**Reply by Ernest-su** [reply]
![录屏_VirtualBox Machine_20220831115124](https://user-images.githubusercontent.com/5917446/187588961-720e945c-4e1e-4455-b182-5ba5ecc6a3a4.gif)
我apt install chromium 用chromium测试，也是一样的。这些浏览器打开文件的界面，用的都是系统内置的文件管理器，系统是我新安装的，没装过第三方的文件管理器，所以不明白为什么你说这个调起的文件打开对话框并非深度文件管理器提供的

**Reply by BLumia** [reply]
![image](https://user-images.githubusercontent.com/10095765/187589453-8cadea6a-7e18-4b03-be33-59b4530711c1.png)
深度文管的打开文件对话框是这样的（此截图使用 deepin v20.6，使用文管版本 `dde-file-manager 5.6.4-1`）。

![image](https://user-images.githubusercontent.com/10095765/187589617-5013d084-e588-4adf-83eb-5d932f456ab6.png)

另外需要留意这个选项，如果这个选项没开的话，是不会默认使用深度文管的打开文件对话框的。安装了深度文管不代表打开文件的时候一定会调起深度文管提供的打开文件对话框。

**Reply by Ernest-su** [reply]
你的文件管理器版本比较低，我的是今天刚从deepin安装的v23版本，100%确认打开的就是深度的文件管理打开文件对话框。你可以升级文件管理器到我的版本测试。 你说的这个设置我检查了，深度文件管理器默认就是打开的。

**Reply by Ernest-su** [reply]
另外，复现问题麻烦使用提供的复现步骤：

打开百度，点搜索框右边的图片图标
打开文件
在打开的【打开文件页面 按ctrl+L，在地址栏输入文件地址
按回车，无法定位到文件
去掉文件名，按回车，能定位到文件所在文件夹。

**Reply by BLumia** [reply]
> 你的文件管理器版本比较低，我的是今天刚从deepin安装的v23版本，100%确认打开的就是深度的文件管理打开文件对话框。你可以升级文件管理器到我的版本测试。 你说的这个设置我检查了，深度文件管理器默认就是打开的。

v23 版本的源并未支持打开文件对话框的 patch（并非在文管项目进行的，patch 在 ~~[这个项目](https://github.com/linuxdeepin/open-with-dialog)~~ [这个项目](https://github.com/linuxdeepin/dde-file-dialog-gtk/tree/gtk3)），但你截图所示的确实不是深度文管的打开文件对话框。深度文管的打开文件对话框左侧是不会有“其它位置”字样的选项的，你目前所看到/使用的那个是 GTK 的打开文件对话框菜单。

**Reply by BLumia** [reply]
补充：我只是为了纠正 “deepin的文件管理器原来已经支持这个功能了” 的说法，目前深度文管的打开文件对话框确实是不支持这个的。实现这个特性也不难做，但需要先确认需求...

**Reply by Ernest-su** [reply]
好的，期待。
